File: android/react-native-camera-kit/src/main/java/com/rncamerakit/recorder/OnRecorderCallbacks.java
package com.rncamerakit.recorder;

public interface OnRecorderCallbacks {
    default void  onTakePhoto(String photoPath){}
    default void  onProgress(long duration){}
    default void  onComplete(boolean validClip, long clipDuration){}
    default void  onFinish(String outputPath){}
    default void  onError(int errorCode){}
}

File: android/app/src/main/java/com/rncamerakit/recorder/OnRecorderCallbacks.kt
package com.rncamerakit.recorder

abstract class OnRecorderCallbacks {
    open fun onTakePhoto(photoPath: String?) {}
    open fun onProgress(duration: Long) {}
    open fun onComplete(validClip: Boolean, clipDuration: Long) {}
    open fun onFinish(outputPath: String?) {}
    open fun onError(errorCode: Int) {}
}